src/main/scala/io/iohk/ethereum/validators/SignedTransactionValidator.scala

```diff
@@ -1,34 +1,41 @@
 package io.iohk.ethereum.validators
 
-import java.math.BigInteger
-
 import io.iohk.ethereum.crypto.ECDSASignature
-import io.iohk.ethereum.domain.{Address, SignedTransaction, Transaction}
-import io.iohk.ethereum.validators.SignedTransactionError.{TransactionSignatureError, TransactionSyntaxError}
-import io.iohk.ethereum.utils.Config
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.validators.SignedTransactionError._
+import io.iohk.ethereum.utils.BlockchainConfig
+import io.iohk.ethereum.vm.{EvmConfig, UInt256}
 
 trait SignedTransactionValidator {
 
-  def validateTransaction(stx: SignedTransaction, fromBeforeHomestead: Boolean): Either[SignedTransactionError, SignedTransaction]
+  def validate(stx: SignedTransaction, senderAccount: Account, blockHeader: BlockHeader,
+               calculateUpfrontGasCost: Transaction => UInt256, accumGasUsed: BigInt): Either[SignedTransactionError, SignedTransaction]
 
 }
 
-object SignedTransactionValidator extends SignedTransactionValidator {
+class SignedTransactionValidatorImpl(blockchainConfig: BlockchainConfig) extends SignedTransactionValidator {
 
   val secp256k1n: BigInt = BigInt("115792089237316195423570985008687907852837564279074904382605163141518161494337")
 
   /**
-    * Validates a transaction
+    * Initial tests of intrinsic validity stated in Section 6 of YP
     *
-    * @param stx                  Transaction to validate
-    * @param fromBeforeHomestead  Whether the block to which this transaction belongs is from
-    *                             before the HomesteadBlock
+    * @param stx                        Transaction to validate
+    * @param senderAccount              Account of the sender of the tx
+    * @param blockHeader                Container block
+    * @param calculateUpfrontGasCost    Function used to calculate the upfront gas cost
+    * @param accumGasUsed               Total amount of gas spent prior this transaction within the container block
     * @return Transaction if valid, error otherwise
     */
-  def validateTransaction(stx: SignedTransaction, fromBeforeHomestead: Boolean): Either[SignedTransactionError, SignedTransaction] = {
+  def validate(stx: SignedTransaction, senderAccount: Account, blockHeader: BlockHeader,
+               calculateUpfrontGasCost: Transaction => UInt256, accumGasUsed: BigInt): Either[SignedTransactionError, SignedTransaction] = {
```

Comment:
why not just `upfrontGasCost: UInt256`? and than pass `calculateUpfrontCost(stx.tx)` in Ledger?